src/main/java/io/anserini/rerank/lib/AxiomReranker.java

```diff
@@ -307,6 +310,11 @@ private ScoredDocuments processExternalContext(ScoredDocuments docs, RerankerCon
     throws IOException {
     Set<Integer> docidSet = new HashSet<>(Arrays.asList(ArrayUtils.toObject(
       Arrays.copyOfRange(docs.ids, 0, Math.min(this.R, docs.ids.length)))));
+
+    if (this.useCollection) {
```

Comment:
Not sure why you return with only `R` documents in the working set?
I think we should keep `selectDocs` untouched to get `R*N` documents as the working set? 
I think what @lintool proposed is another way to compute the MI, but not how to construct the working set?